\documentclass[a4paper]{article}
\usepackage{Rd}
% \VignetteIndexEntry{STRING data-package-generator}
% #\VignetteDepends{AnnotationDbi}

\title{R-STRING.db - building STRING datapackages}
\author{Stefan McKinnon Edwards \\ \email{stefan.hoj-edwards@@agrsci.dk} \\ Center of Quantitative Genetics and Genomics, Aarhus University}
\date{\today}

\begin{document}
%\SweaveOpts{concordance=TRUE}

\maketitle
\tableofcontents


\section{Introduction}
The STRING database \cite{STRING2011,STRING8} is a very large repository of
known and predicted protein-protein interactions.
Currently, STRING version 9.0 covers 5'214'234 proteins from 1133 organisms.
The website (\url{http://string-db.org/}) contains search features for looking up proteins of interest and the data is even downloadable for a more systematic analysis.

However, the most simple of files is a 2.7 Gb gzipped compressed file and contains every protein-protein link for all 1133 organisms.
Reading this file, extracting the lines for the organism in question and loading the data into a meaningful data structure can be done in many ways,
and routines for this is likely passed on from project to project, slowely evolving.
A data package, like \code{org.Bt.eg.db} on BioConductor \cite{BioC}, has the advantages of being available on the system and provides a consistent use. 
This package provides a frame-work for indexing the large file, 
extracting the data specific for a organism and repacking it into a datapackage, including mapping the ENSEMBL protein identifiers into several other identifiers.

\subsection{The data}
Let us take a look at the input data.
We use the file \texttt{protein.links.v9.0.txt.gz} (named so for STRING v. 9.0) which just lists the proteins 
and a combined score for the link. 
The links are scored on several channels (e.g.\ experimental evidence, prediction and more) and these can be found in the detailed file.

The first couple of lines of the first mentioned file looks like this:
\begin{verbatim}
      protein1       protein2  combined_score
394.NGR_a00010 394.NGR_a00020             912
394.NGR_a00010 394.NGR_a00050             854
394.NGR_a00010 394.NGR_a00070             175
394.NGR_a00010 394.NGR_a00080             626
\end{verbatim} 
and some 26 mio lines later, we find the human links data:
\begin{verbatim}
9606.ENSP00000000233 9606.ENSP00000158762 619
9606.ENSP00000000233 9606.ENSP00000174653 266
9606.ENSP00000000233 9606.ENSP00000203407 296
\end{verbatim}

The protein identifiers is made up of the pattern \texttt{nnnn.aaaaa} where \texttt{nnnn} is the 
NCBI taxonomy species identifier\footnote{\url{http://www.ncbi.nlm.nih.gov/taxonomy}} 
and \texttt{aaaaa} is the RefSeq/Ensembl-identifier of the protein.

\section{Usage}

\subsection{Step 1: Indexing and extracting}

\subsection{Step 2: Building a new package}

\section{What was used to create this document}
The version number of R and the packages and their versions that were used to generate this
document are listed below.
<<>>=
sessionInfo()
@

\section{References}

\begin{thebibliography}{10}

\bibitem{BioC}
Robert~C Gentleman, Vincent~J. Carey, Douglas~M. Bates, et~al.
\newblock Bioconductor: Open software development for computational biology and
  bioinformatics.
\newblock {\em Genome Biology}, 5:R80, 2004.

\bibitem{STRING2011}
Damian Szklarczyk,
Andrea Franceschini, 
Michael Kuhn, 
Milan Simonovic, 
Alexander Roth, 
Pablo Minguez, 
Tobias Doerks, 
Manuel Stark, 
Jean Muller, 
Peer Bork, 
Lars~J Jensen, 
Christian~von~Mering, 
\newblock The STRING database in 2011: functional interaction networks of proteins, globally integrated and scored.
\newblock {\em Nucleic acids research}, (2011) Jan;39(Database issue):D561-8.

\bibitem{STRING8}
Benjamin Lehne and Thomas Schlitt
\newblock Protein-protein interaction databases: keeping up with growing interactomes. 
\newblock {\em Human genomics 3}, 291???7 (2009).

\end{thebibliography}

\end{document}